vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -105,7 +105,7 @@ void AdsImpl::InitializeStep3() {
 }
 
 void AdsImpl::Deinitialize() {
-  if (!IsInitialized()) {
```

Comment:
It's a little awkward that checking the flag versus calling `IsInitialized` have different outcomes. Seems like the real problem here is that `IsInitialized` is also checking the enabled state. I guess that would be tough to rearchitect though, if clients have become reliant on that behavior.

Reply:
Let me have a think about this as a very valid point sir